src/sort.js

```diff
@@ -0,0 +1,73 @@
+var common = require('./common');
+var fs = require('fs');
+
+//@
+//@ ### sort([options,] file [, file ...])
+//@ ### sed([options,] file_array)
+//@ Available options:
+//@
+//@ + `-r`: Reverse the result of comparisons
+//@ + `-n`: Compare according to numerical value
+//@
+//@ Examples:
+//@
+//@ ```javascript
+//@ sort('foo.txt', 'bar.txt');
+//@ sort('-r', 'foo.txt');
+//@ ```
+//@
+//@ Description -- todo
+function _sort(options, files) {
+  options = common.parseOptions(options, {
+    'r': 'reverse',
+    'n': 'numerical'
+  });
+
+  // Check if this is coming from a pipe
+  var pipe = common.readFromPipe(this);
+
+  if (!files && !pipe)
+    common.error('no files given');
+
+  files = [].slice.call(arguments, 1);
+
+  if (pipe)
+    files.unshift('-');
+
+  var lines = [];
+  files.forEach(function(file) {
+    if (!fs.existsSync(file) && file !== '-') {
+      // exit upon any sort of error
+      common.error('no such file or directory: ' + file);
+    }
+
+    var contents = file === '-' ? pipe : fs.readFileSync(file, 'utf8');
+    lines = lines.concat(contents.trimRight().split(/\r*\n/));
+  });
+
+  var sorted;
+  function parseNumber (str) {
+    var match = str.match(/^\s*(\d*)\s*(.*)$/);
+    return {num: Number(match[1]), value: match[2]};
+  }
+  if (options.numerical) {
+    sorted = lines.sort(function(a, b) {
+      var objA = parseNumber(a);
+      var objB = parseNumber(b);
+      if (objA.hasOwnProperty('num') && objB.hasOwnProperty('num')) {
+        return (objA.num !== objB.num) ? (objA.num - objB.num)
+            : (objA.value.localeCompare(objB.value));
+      } else {
+        return (objA.value.localeCompare(objB.value));
```

Comment:
It's the default string comparison. It returns `-1` if `this` is "less than" (sorted before) the argument.

So:

```
'foo'.localeCompare('bar'); // 1
'foo'.localeCompare('zed'); // -1
'foo'.localeCompare('foo'); // 0
```